pulp_container/app/registry_api.py

```diff
@@ -1117,9 +1121,13 @@ def fetch_manifest(self, remote, pk):
                 # the client could request the manifest outside the docker hub pull limit;
                 # it is necessary to pass this information back to the client
                 raise Throttled()
-            else:
-                # TODO: do not mask out relevant errors, like HTTP 502
+            elif response_error.status == 404:
                 raise ManifestNotFound(reference=pk)
+            else:
+                raise BadGateway(detail=response_error.message)
+        except (ClientConnectionError, TimeoutException):
+            # The remote server is not available at the moment
+            raise GatewayTimeout()
```

Comment:
can you also pass the error message here in the detail?

Reply:
`TimeoutException` does not have an attribute for the error message: https://github.com/pulp/pulpcore/blob/778108af02e75a44d53a66749529e1b0acd751c7/pulpcore/exceptions/base.py#L71.

The same goes for `ClientConnectionError`: https://docs.aiohttp.org/en/stable/_modules/aiohttp/client_exceptions.html#ClientConnectionError.

I would leave this part untouched. Can we agree on that?